### PR TITLE
fix source from last predicate

### PR DESCRIPTION
**Pull request by szuecs**
fix source from last predicate was based on wrong assumptions, we do not get parsed headers into string slice, but a plain string, so use Header.Get and work on the string ourselves

reported by a user

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

### CODE REVIEW

**Comment by szuecs** [comment]
:+1:

File: net/net.go [hunk 1]
```diff
@@ -50,16 +50,14 @@ func RemoteHost(r *http.Request) net.IP {
...
-	l := len(a) - 1
-	if l < 0 {
-		l = 0
-	}
-	if ffh := parse(a[l]); ffh != nil {
-		return ffh
+	ffs := r.Header.Get("X-Forwarded-For")
+	ffa := strings.Split(ffs, ",")
+	ff := ffa[len(ffa)-1]
+	if ff != "" {
```
Comment (aryszka) [inline net/net.go]: i think it would be better to trim the space before checking for empty. What do you think?

**Comment by aryszka** [comment]
:+1:

File: net/net.go (hunk 1 above)
Comment (szuecs) [reply net/net.go]: We need to trim the words as far as I tested these have `, `

**Comment by szuecs** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Comment by aryszka** [comment]
:+1:

**Pull request merged**